src/dotnet/commands/dotnet-nuget/NuGetCommand.cs

```diff
@@ -0,0 +1,85 @@
+// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Reflection;
+using Microsoft.DotNet.Cli.CommandLine;
+using Microsoft.DotNet.Cli.Utils;
+using Microsoft.DotNet.InternalAbstractions;
+using NugetProgram = NuGet.CommandLine.XPlat.Program;
+
+namespace Microsoft.DotNet.Tools.NuGet
+{
+    public class NuGetCommand
+    {
+        private static readonly string DefaultRid = RuntimeEnvironmentRidExtensions.GetLegacyRestoreRuntimeIdentifier();
+
+        public static int Run(string[] args)
+        {
+            DebugHelper.HandleDebugSwitch(ref args);
+
+            var app = new CommandLineApplication(false)
+            {
+                Name = "dotnet nuget",
+                FullName = ".NET nuget command passthrough",
+                Description = "Passes selected commands to the NuGet layer"
+            };
+
+
+            app.OnExecute(() =>
+            {
+                try
+                {
+                    return RunCommand(args, new NuGetCommandRunner());
+                }
+                catch (InvalidOperationException e)
+                {
+                    Console.WriteLine(e.Message);
+
+                    return -1;
+                }
+                catch (Exception e)
+                {
+                    Console.WriteLine(e.Message);
+
+                    return -2;
+                }
+            });
+
+            return app.Execute(args);
+        }
+
+        public static int RunCommand(IEnumerable<string> args, INuGetCommandRunner commandRunner)
+        {
+            Debug.Assert(commandRunner != null, "A command runner must be passed to RunCommand");
+            if (commandRunner == null)
+            {
+                return -1; // invalid operation
+            }
+
+            string[] commandArgs = args.ToArray();
+            if (commandArgs.Length == 0 ||
+                !commandArgs[0].Equals("locals", StringComparison.OrdinalIgnoreCase) &&
+                !commandArgs[0].Equals("push", StringComparison.OrdinalIgnoreCase) &&
+                !commandArgs[0].Equals("delete", StringComparison.OrdinalIgnoreCase))
+            {
+                return 1; // unsupported argument
```

Comment:
Good catch--thanks. Here is my plan--I'll go with option 2, so that we can have a "single point of truth" for the usage text, NuGet.CommandLine.XPlat.dll. We also have a Windows exe that invokes the command in this dll, so this would avoid any duplication. This way I also won't need to do any modifying of options before passing them to the NuGet dll, as <command> <option> is its argument format too. For example "dotnet nuget push --help" would be passed to the NuGet dll as "push --help", and we can light up usage text appropriately from there.